### ISSUE TITLE
Incorrect group choice parsing

### ISSUE THREAD

**Issue by rooooooooob** [anweiss/cddl#11]
Take the following code as an example:

```cddl
foo = int

; no errors
bar = { int, int // int, tstr }

; error
; baz = { int, foo // int, tstr }
```

It seems to happen whenever a typename is used as the last field within a group choice.
It generates an Error return. However, even in the first example, `bar`, I think it is still incorrectly parsing, as it seems to consider it more equivalent to what `bar = {int, int int, tstr }` might be, although that in and of itself seems weird.

As per the RFC8610 specs:

> Analogous to types, CDDL also allows choices between groups,
   delimited by a "//" (double slash).  Note that the "//" operator
   binds much more weakly than the other CDDL operators, so each line
   within "delivery" in the following example is its own alternative in
   the group choice:

                   address = { delivery }

                   delivery = (
                   street: tstr, ? number: uint, city //
                   po-box: uint, city //
                   per-pickup: true )

                   city = (
                   name: tstr, zip-code: uint
                   )

`bar` should be parsed into 2 group choices with 2 members each, not 1 choice with 4 members.

Here is the debug-printed value of `bar`:
```rust
CDDL { rules: [Type(TypeRule {
    name: Identifier(("bar", None)),
    generic_param: None,
    is_type_choice_alternate: false,
    value: Type([
        Type1 {
            type2: Map(Group([
                GroupChoice([
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("int", None)), None)), operator: None }]) }), true),
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("int", None)), None)), operator: None }]) }), false),
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("int", None)), None)), operator: None }]) }), true),
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("tstr", None)), None)), operator: None }]) }), false)
                ])
            ])),
            operator: None
        }
    ]),
    range: (0, 7)
})] }
```

And here is how `bar = { int, int, int, tstr }` is parsed:
```rust
CDDL { rules: [Type(TypeRule {
    name: Identifier(("bar", None)),
    generic_param: None,
    is_type_choice_alternate: false,
    value: Type([
        Type1 {
            type2: Map(Group([
                GroupChoice([
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("int", None)), None)), operator: None }]) }), true),
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("int", None)), None)), operator: None }]) }), true),
                    (ValueMemberKey(ValueMemberKeyEntry { occur: None, member_key: None, entry_type: Type([Type1 { type2: Typename((Identifier(("tstr", None)), None)), operator: None }]) }), false)
                ])
            ])),
            operator: None
        }]),
        range: (0, 7)
})] }
```
Which is missing one of the fields, possibly since they are unnamed?

Which made me try `bar = ( a: int, b: int // c: int, d: tstr )` but it still has the same problem as before with them all being in one group choice.

**Comment by anweiss** [comment]
Thanks for reporting @rooooooooob. Will dig into this and attempt to address.

**Comment by SebastienGllmt** [comment]
Hey @anweiss thanks for offering to help! We'd love to help get this library working and used in production. Would you be able to get on contact with us? You can email me at sebastien [at] emurgo.io

**Comment by anweiss** [comment]
Hey @SebastienGllmt, just shot you a note via email.

**Comment by anweiss** [comment]
This should be fixed via #12. Closing.
